log4j-core/src/main/java/org/apache/logging/log4j/core/util/Throwables.java

```diff
@@ -40,12 +40,26 @@ private Throwables() {
      * @return the deepest throwable or the given throwable
      */
     public static Throwable getRootCause(final Throwable throwable) {
+
```

Comment:
Seems like there's a broken test related to this change?

Reply:
There was a code typo. Fixed it and added a test to check against causal chains with loops.